src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -47,32 +47,67 @@ public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, stri
                     return null;
                 }
                 // "ClCompile" for source files and "ClCompile" for header files
-                if (file.ItemType != "ClCompile")
+                // Don't rely on file extension to indentify header files since they can have any extension
+                var itemType = file.ItemType;
+                if (itemType != "ClCompile" && itemType != "ClInclude")
                 {
                     logger.WriteLine($"File's \"Item type\" is not supported. File: '{absoluteFilePath}'");
                     return null;
                 }
                 var platformName = ((VCPlatform)vcConfig.Platform).Name; // "Win32" or "x64"
                 var vcFileConfig = file.GetFileConfigurationForProjectConfiguration(vcConfig);
-                // We don't support custom build tools. VCCLCompilerTool is needed for all the necessary compilation options to be present
-                if (!(vcFileConfig.Tool is VCCLCompilerTool))
+                IVCRulePropertyStorage vcFileSettings;
+                if (itemType == "ClCompile")
                 {
-                    logger.WriteLine($"Custom built files are not supported. File: '{absoluteFilePath}'");
-                    return null;
+                    // We don't support custom build tools. VCCLCompilerTool is needed for all the necessary compilation options to be present
+                    if (!(vcFileConfig.Tool is VCCLCompilerTool))
+                    {
+                        logger.WriteLine($"Custom built files are not supported. File: '{absoluteFilePath}'");
+                        return null;
+                    }
+                    // The Tool property is typed as Microsoft.VisualStudio.VCProjectEngine.VCCLCompilerTool, and
+                    // we could use it to fetch most of the properties we need, rather than fetching them via the
+                    // IVCRulePropertyStorage interface. However, not all of the properties are exposed directly by
+                    // VCCLCompilerTool (e.g. LanguageStandard). Also, quite a few of the Tool properties are exposed
+                    // as enums, so we'd need to change our code to handle them.
+                    vcFileSettings = vcFileConfig.Tool as IVCRulePropertyStorage;
+                }
+                else
+                {
+                    // header file load file setting from the project
+                    var tools = vcConfig.Tools as IVCCollection;
+                    var toolItem = tools.Item("VCCLCompilerTool");
+                    if (toolItem == null)
+                    {
+                        logger.WriteLine($"Custom built files are not supported. File: '{absoluteFilePath}'");
```

Comment:
This is a repetition, please see my comment above regarding the method: with a dedicated method, it will be easier to do something like
```
object toolItem;
if (itemType == "ClCompile") 
{
   toolItem = ...
} 
else
{
   toolItem = ...
}

if (toolItem == null) 
{
   logger.WriteLine...
   return null;
}
return toolItem as IVCRulePropertyStorage;
```